### ISSUE TITLE
emails we send from Sendgrid are being blocked by many organizations

### ISSUE THREAD

**Issue by hwatheod** [learning-unlimited/ESP-Website#3105]
Today I noticed that emails from the dev3 server to my cs.stanford.edu account were not received.  Then I checked our sendgrid account.  The email was sent out by sendgrid, but was blocked by the cs.stanford.edu server. It's not even putting it in spam, it's just blocking it completely.  This could be a significant problem since we likely have many users from that domain.

From sendgrid records, emails have been blocked recently by cs.stanford.edu, yale.edu, hotmail.com, msn.com and many other domains. We should try to resolve the issue with Sendgrid. The IP in the mssage below is a Sendgrid IP.

Sendgrid says:

The cs.stanford.edu server blocked this message from being delivered. While we won't try to send this message again, we will attempt to send new messages to this address in the future. 

Full response from the cs.stanford.edu server:

550 Mail from 167.89.100.175 is blacklisted. Please see http://www.spamhaus.org/query/bl?ip=167.89.100.175 for further information.

**Comment by willgearty** [comment]
I think we noticed this a bit during Rainstorm. There was also a related but different issue (I think) where our server was flat out rejecting emails from particular university domains (I think MIT was one of them). In both cases, the temporary solution we used then was to just suggest that teachers use personal email addresses (gmail, preferably).

Following that link, it looks like we are using the same sendgrid IP address as some other sendgrid users that are sending spam. I guess this was an inevitable possibility on our particular sendgrid plan, and it's possible that we could reach out to sendgrid to resolve the issue (at least temporarily). I don't know if sending spam through sendgrid is against the TOS. If so, they could just ban those users and resolve the issue with the blacklist, but if not, maybe we could at least request a different shared IP address.

**Comment by willgearty** [comment]
OK, I've submitted a ticket to Sendgrid for the 3 IP addresses that we've been getting block messages for. From their support documentation, it sounds like this happens a bit, and while it sounds like they deal with it fairly quickly, we also probably want to be diligent about monitoring for this. In that vein, I've set up a weekly report email to the web-team address with a summary of requests, deliveries, bounces, etc.

On the bright side, over the last week we sent nearly 19,000 emails and >90% of them were delivered and only 2% of them were blocked.
